File: CurrencyExchange.API/CurrencyExchange.Services/Dto/ConversionRateHistoryDto.cs
using System.Collections.Generic;
using CurrencyExchange.Services.Enums;

namespace CurrencyExchange.Services.Dto
{
    public class ConversionRateHistoryDto
    {
        public string FromCurrency { get; set; }
        public string ToCurrency { get; set; }
        public IEnumerable<DailyConversionRate> DailyConversionRates { get; set; } = new List<DailyConversionRate>();
        public ErrorCode ErrorCode { get; set; }
    }
}

File: CurrencyExchange-UI/src/app/shared/dtos/conversion-rate-history.dto.ts
import { DailyConversionRate } from '../models/daily-conversion-rate.model';

export class ConversionRateHistoryDto {
  constructor(
    public fromCurrency: string,
    public toCurrency: string,
    public dailyConversionRates: Array<DailyConversionRate>,
    public errorCode: number
  ) {}
}
